src/backend/alsa/mod.rs

```diff
@@ -143,6 +143,12 @@ pub struct MidiInputPort {
     addr: Addr,
 }
 
+impl MidiInputPort {
+    pub fn id(&self) -> String {
+        format!("{}..{}", self.addr.client, self.addr.port)
```

Comment:
I'm curious what's the reason for this specific format with the two dots? Is there precedence for this somewhere?

Reply:
This can be broken into two parts.

## Why a separator?

Lets pretend the numbers for one port are `11` and `1` and another port is `1` and `11`, you will see that when we concat both of these they both result in the same string `111` which would mean the port id is not always unique which is a major problem. To solve this we can put in a separator which consists of a character that can never appear in a `i32`.

## Why that separator?

I choose `..` for no particular reason.

Reply:
I think a `:` is usually used to separate client and port in ALSA, and makes more sense to me than `..`. Could you change that? Then I think this PR is good to merge 👍